app/assets/javascripts/hubstats/application.js

```diff
@@ -14,45 +14,85 @@
+//= require bootstrap-datepicker
 //= require_tree .
 
 $(document).ready( function() {
+  $('.input-daterange').datepicker({
+    "todayHighlight": true,
+    "endDate": "Today",
+    "todayBtn": true
+  });
   setDateRange()
 });
 
 function setDateRange() {
-  var index = readCookie("hubstats_index") || 2;
-  var timer = document.getElementById("time-select");
+  var dates;
+  var timer;
+  var datesArray = [];
 
-  timer.selectedIndex = index;
+  if ((readCookie("hubstats_dates") === "null~~null") || (readCookie("hubstats_dates") === null)) {
+    dates = getDefaultDateRange();
+  } else {
+    dates = readCookie("hubstats_dates")
+  }
 
-  timer.onchange = function() {
-    createCookie("hubstats_index",this.selectedIndex,1);
+  timer = document.getElementById("submitDateRange");
+  datesArray = dates.split("~~")
+
+  $('.input-daterange').find('[name="start"]').datepicker('update', new Date(datesArray[0]));
+  $('.input-daterange').find('[name="end"]').datepicker('update', new Date(datesArray[1]));
+
+  timer.onclick = function() {
+    var start_date;
+    var end_date;
+    var date;
+    start_date = $('.input-daterange').find('[name="start"]').datepicker('getDate');
+    end_date = $('.input-daterange').find('[name="end"]').datepicker('getDate');
+    
+    date = new Date();
+    date.setTime(date.getTime()+(24*60*60*1000));
+    var expires = "; expires="+date.toGMTString();
+    createCookie("hubstats_dates=" + start_date + "~~" + end_date + expires + "; path=/");
     window.location.reload();
   };
 };
 
-function createCookie(name,value,days) {
-    if (days) {
-        var date = new Date();
-        date.setTime(date.getTime()+(days*24*60*60*1000));
-        var expires = "; expires="+date.toGMTString();
-    }
-    else var expires = "";
-    document.cookie = name+"="+value+expires+"; path=/";
+function createCookie(str) {
+  document.cookie = str;
 };
 
 function readCookie(name) {
-    var nameEQ = name + "=";
-    var ca = document.cookie.split(';');
-    for(var i=0;i < ca.length;i++) {
-        var c = ca[i];
-        while (c.charAt(0)==' ') c = c.substring(1,c.length);
-        if (c.indexOf(nameEQ) == 0) return c.substring(nameEQ.length,c.length);
-    }
-    return null;
+  var cookieName = name + "=";
+  var cookieData = document.cookie.split(';');
+  for(var i=0;i < cookieData.length;i++) {
+    var c = cookieData[i];
+    while (c.charAt(0)==' ') c = c.substring(1,c.length);
+    if (c.indexOf(cookieName) == 0) return c.substring(cookieName.length,c.length);
+  }
+  return null;
 };
 
 function eraseCookie(name) {
-    createCookie(name,"",-1);
+  createCookie(name,"",-1);
+};
+
+function getDefaultDateRange() {
+  var currentDate = new Date();
+  var dd = currentDate.getDate();
+  var mm = currentDate.getMonth() + 1; //January is 0!
+  var yyyy = currentDate.getFullYear();
+
+  if(dd < 10) {
+    dd = '0' + dd
+  } 
+
+  if(mm < 10) {
+    mm = '0' + mm
+  }
+
+  todaysDate = new Date(mm + '/' + dd + '/' + yyyy);
```

Comment:
We need the format mm/dd/yyyy so that it knows what to put into the datepicker. The `twoWeeksAgo` already has a built in form like this if we build the new date on today's date.